data/mods/gen9ssb/moves.ts

```diff
@@ -2254,6 +2254,61 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		type: "Psychic",
 	},
 
+	// Two of Roses
+	dillydally: {
+		accuracy: 90,
+		basePower: 40,
+		category: "Physical",
+		shortDesc: "Hits twice, each hit raises a random stat by 1. User's secondary type determines type.",
+		name: "Dilly Dally",
+		pp: 20,
+		priority: 0,
+		multihit: 2,
+		volatileStatus: 'dillydally',
+		flags: {contact: 1},
+		type: "???",
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onModifyType(move, pokemon) {
+			let type = "???";
+			for (let i = 2; i >= 0; i--) {
+				type = pokemon.getTypes()[i];
+				if (!type || type === undefined) {
+					continue;
+				} else {
+					break;
+				}
+			}
+			if (type === "Bird") type = "???";
+			if (type === "Stellar") type = pokemon.getTypes(false, true)[1];
+			move.type = type;
+		},
```

Comment:
You can use ``let type = pokemon.getTypes()[pokemon.getTypes().length - 1];`` to get a similar effect

Reply:
noted, will do